### ISSUE TITLE
Mojave dark mode causes a white line at the top of Electron apps

### ISSUE THREAD

**Issue by LoganDark** [electron/electron#13164]
* Electron Version: ???
* Operating System (Platform and Version): macOS 10.14 Mojave
* Last known working Electron version: ???

**Expected Behavior**
No white lines

**Actual behavior**
There is a thin white line at the top border of Discord

**To Reproduce**
Download Discord and run.

**Screenshots**
![image](https://user-images.githubusercontent.com/4723091/41010254-329960e0-68eb-11e8-9c92-a62f3dd2006c.png)

**Additional Information**
May be related to dark theme.

---

I am aware macOS Mojave is not exactly accessible to everyone yet, but I'm reporting this just in case.

**Comment by robjtede** [comment]
Confirming this happens in all the Electron apps I have, in Mojave dark mode.

**Comment by zinne** [comment]
tested Mojave yesterday, also seen the line. Also the standard file open dialog appeared still in light

**Comment by ivanreese** [comment]
This happens in both dark mode and light mode. I believe it has something to do with apps that hide the title bar.

I have the issue in Atom, Hyper, and Slack.

**Comment by luizfonseca** [comment]
+1 Even chrome on Mojave has it, so I think this is around chromium/electron apps.

**Comment by samuelgozi** [comment]
I believe that this line is added by the OS, it is added to other apps too, however when the app doesn't use the native title bar then there is no "gradient" and it stands out more because the contrast is higher.
This line also exists in the light UI, but it's less disturbing because the contrast is lower.
I reported this as an "unexpected behavior" to apple but I'm not sure that they will do anything about it(unless many other people complain).

**Comment by LoganDark** [comment]
Built-in About this Mac:
![image](https://user-images.githubusercontent.com/4723091/42418525-14f87df2-8257-11e8-81d2-83657e778b8e.png)
It's not pure white like in Electron.

**Comment by samuelgozi** [comment]
@LoganDark It seems like you are correct, I just checked with 4 different apps, 3 of them are native, and each one of them has a different color on that line.
I checked with "settings", "safari", chrome(which has the same color) and "feedback assistant"

**Comment by gilad215** [comment]
Any update on this? @ivanreese

**Comment by ivanreese** [comment]
@gilad215 It seems this is a Chrome thing? I don't know, I'm just a user of Electron apps, not a dev on any of these projects.

**Comment by jeroen-meijer** [comment]
Nothing has changed after Chrome update 69. This is still an issue.

**Comment by alihaider96** [comment]
Hope it gets resolved somehow, it's really bugging me out 😆
![image](https://user-images.githubusercontent.com/9122088/45553266-51c88380-b82b-11e8-8a1b-ead6fb027550.png)

**Comment by rdiazv** [comment]
Even a transparent window displays the white border 😞

<img width="807" alt="captura de pantalla 2018-09-24 a la s 20 00 56" src="https://user-images.githubusercontent.com/1670771/45983791-924aad00-c034-11e8-9034-eafd22f9ed74.png">

In this case a fullscreen transparent window.

**Comment by glekner** [comment]
Really frustrating. They have to make an option to disable it.

**Comment by npezza93** [comment]
Experiencing this as well.

**Comment by mettaworldjon** [comment]
Yes this issue is happening on both dark and light mode

**Comment by jeroen-meijer** [comment]
Did the official release of Mojave fix the issue?

**Comment by bpasero** [comment]
I am still seeing it with VSCode (https://github.com/Microsoft/vscode/issues/54838)

**Comment by SinarPandora** [comment]
Please use: defaults write -app <application name> NSRequiresAquaSystemAppearance -bool No
It works!

**Comment by Neofish22** [comment]
The solution from @SinarPandora certainly does reduce the white line to a thinner grey one (which I don't remember being there before). Confirmed with Sublime\ Text and Google\ Chrome.

**Comment by karlhorky** [comment]
And `[application name]` is written as just the name, with backslashes before spaces:

```
defaults write -app Visual\ Studio\ Code NSRequiresAquaSystemAppearance -bool No
```

Thanks @SinarPandora!

**Comment by bkeating** [comment]
Keep reaching to resize my window up a pixel. Makes it feel like I can see other window content in the background.

I wonder how other already-dark-by-default apps have fared with the move to system-level dark mode support. Tricky. I don't want macOS touching my Atom electron app at all. I love it just the way it is.

**Comment by romanmarch** [comment]
**I have a border in frame + transparent**

<img width="642" alt="frame" src="https://user-images.githubusercontent.com/36361469/46036943-7a942700-c10f-11e8-8612-a42014fcdc56.png">

**Comment by jeroen-meijer** [comment]
@bkeating
Your electron apps should be fine. Mac doesn’t do anything special when rendering applications in Mojave compared to before. Apps have the ability to check whether dark mode is enabled, but Mac OS renders apps just like before (except for the border issue discussed here and some irrelevant text rendering changes).

**Comment by SinarPandora** [comment]
> **Edit:** Yeah, @Neofish22 is right, it doesn't appear to fix the issue completely :( There is now a new dimmer line around the whole window.
> 
> If you still want to try it, `[application name]` is written as just the name of the application, with backslashes before spaces:
> 
> ```
> defaults write -app Visual\ Studio\ Code NSRequiresAquaSystemAppearance -bool No
> ```

After all, this is an unofficial temporary solution. 
And don't use it on lighter apps.😂

**Comment by texelate** [comment]
I'm not using dark mode and am still getting the issue in Electron apps (Wavebox and Atom). Neither are using the system title bar.

**Comment by vanejung** [comment]
@texelate If the native title bar buttons (close, minimise ..) are present, the white line seems to be visible. Perhaps the white line is a part of the title bar buttons. Not 100% sure.

**Comment by m4kamran** [comment]
I regret upgrading. That white line is so irritating.

**Comment by gavJackson** [comment]
I have this issue too, doesn't matter if OSX is in dark or light mode tho: Looks pretty bad on my app too - not as bad as it does on menu bar apps like @evilrabbit 

<img width="476" alt="screenshot 2018-09-27 at 08 58 12" src="https://user-images.githubusercontent.com/164279/46131459-8ed14480-c233-11e8-911f-3649abed029f.png">

**Comment by pronebird** [comment]
`NSRequiresAquaSystemAppearance` does not seem to work at all. The hairline at the top is visible in both dark and light modes.

**Comment by morajabi** [comment]
I have it in my Chrome too. (I'm using a dark theme, but the white border looks identical to what I see in VS Code, Hyper, Now, There)
![cleanshot 2018-09-27 at 18 36 50 2x](https://user-images.githubusercontent.com/12202757/46155500-4f7f1400-c284-11e8-8003-bc6d423486fd.png)

**Comment by antonreshetov** [comment]
We need to sign the petition to Apple has removed that line 😀 but for me it is a utopia 🤪, will have to live with it...

**Comment by samuelgozi** [comment]
I don't think that this is an Electron/MacOS issue, I think it might be a Chromium issue.
The only non-Electron app that has this issue is Chrome, and that is what leads me to that conclusion.
To make things clear, MacOS Mojave does add a light line to the window, but its lighter than the one that we see in the current electron/chrome apps, and it changes depending on the position of the window. So the closer it is to the top the lighter it gets, but it does look better than the current one.

**Comment by vanejung** [comment]
@morajabi @samuelgozi I don't think that it is a Chromium nor Electron side issue. You can see the same white line issue happening on native apps as well such as TextEdit, Terminal, etc.

<img width="750" alt="screen shot 2018-09-29 at 10 31 46 am" src="https://user-images.githubusercontent.com/29095096/46238800-80219380-c3d3-11e8-8ad1-bf51c80a51d4.png">

**Comment by pronebird** [comment]
The white line is not the issue, it’s by design in Mojave. The issue is that transparent window has it rendered in the wrong place (unlike shadow).

**Comment by samuelgozi** [comment]
I tested it various times with color pickers and different positions, the line is there by design, but in chromium apps(electron apps and chrome) the line is way lighter, in other apps the line is darker and changes based on the position of the window, so the higher the window the lighter the line, but its never as light as in chromium apps.

That's why I think its a chromium issue. If chrome had a darker or no line like the native apps, that would be a big hint that it's an electron problem, but for now, im pretty sure its a chromium problem.

**Comment by vanejung** [comment]
@morajabi @samuelgozi Noticed that non-Electron apps such as Sketch and Affinity Photo have the identical prominent white line issue along with Chrome. Again, this seems to be an OS side problem, not just Chromium. By the way, oddly, some other apps such as Finder, TextEdit and Spark have a different window frame treatment. Maybe there are frame options or modes for developers to choose? That I'm not too sure.

Comparison - Sketch & Chrome
<img width="588" alt="screen shot 2018-09-30 at 9 45 08 am" src="https://user-images.githubusercontent.com/29095096/46251441-10c0a800-c496-11e8-9ecc-8c354260c0c9.png">

**Comment by glekner** [comment]
Does anyone experience a performance improvement or any significant change that makes him wanna stay?

I'm actually considering degrading to HS cause of this unbearable white border

**Comment by axelkennedal** [comment]
I've reported this bug to Apple, I suggest you'd do the same to get their attention, https://www.apple.com/feedback/macos.html

**Comment by MAPESO** [comment]
@vanejung The sketch application is built in electron.

**Comment by MAPESO** [comment]
This problem is `Chromium`. The applications that are **not built with electron**, do not present this horrible white border

<img width="1280" alt="screen shot 2018-10-05 at 00 42 05" src="https://user-images.githubusercontent.com/16585386/46518077-22100880-c838-11e8-87e7-a52682eaee46.png">

**Comment by morajabi** [comment]
@MAPESO Are you sure Sketch and Affinity are in Electron or Chromium? Given the size of the application it's a bit unbelievable but please do share if you have evidence?

**Comment by Scotttresor** [comment]
hello i wont to subscrie

**Comment by hemlok** [comment]
@MAPESO @morajabi sketch is a native app

**Comment by morajabi** [comment]
To summarize the thread, *it's definitely a "bug" and not intended to be in Mojave dark mode.*

Where is the bug, we're not sure:

- it might be Chromium only (probably not)
- A macOS native thing that Chromium got it wrong (causing all Electron apps to have it) alongside with couple other native apps like Sketch or Affinity. 
If they identify the issue, we'll get rid of this terrible border.

**Comment by axelkennedal** [comment]
> hello i wont to subscrie

@Scotttresor As long as it's not to T-Series then you have my permission.

**Comment by MAPESO** [comment]
@morajabi @hemlok I just checked the information. I was wrong, I'm sorry for the confusion. Sketch is a native app ❤️

**Comment by MAPESO** [comment]
@morajabi We could say that it is a problem `Chromium` and `macOS Mojave`

**Comment by ioncodes** [comment]
Reported this to Apple as well, the line is okay on macOS apps such as Finder, but it's really annoying on third party apps such as Chrome, VSCode, etc.

**Comment by itsandreramon** [comment]
Happens on other apps such as WhatsApp or Spotify too, which are not Electron Apps.

**Comment by MAPESO** [comment]
@itsandreramon Spotify uses [`Chromium`](https://www.quora.com/How-is-JavaScript-used-within-the-Spotify-desktop-application-Is-it-packaged-up-and-run-locally-only-retrieving-the-assets-as-and-when-needed-What-JavaScript-VM-is-used), for that reason they are experiencing the same problem as the electron applications.

**Comment by saadq** [comment]
@itsandreramon I'm pretty sure the WhatsApp desktop app is an Electron app.

**Comment by itsandreramon** [comment]
Oh okay, did not knew that. Affects a lot of apps then.. bummer.

**Comment by ioncodes** [comment]
You can find a list of apps made with Electron [here](https://electronjs.org/apps)

**Comment by samuelgozi** [comment]
This is going nowhere.
We need someone that is involved with the Chromium project to help us find out if this is a problem with how Chromium handles windows(maybe it's not using the updated API right) or with the OS.

Since I've already filed bug reports to apple(around 2 months before the release) and nothing happened, we might need to file them to Chromium too.

Here is a link for filing bug reports on Chromium: https://crbug.com/wizard
And here are the "Bug Reporting Guidelines for the Mac & Linux builds": https://www.chromium.org/for-testers/bug-reporting-guidlines-for-the-mac-linux-builds

I strongly believe that this is an issue with Chromium because you can clearly see that there is a difference between native apps and Electron ones, and that leads me to believe that maybe the native ones use the window API differently.

Please report the bug if you have time.

**Comment by itsandreramon** [comment]
Happens on NordVPN too, which is not a Chromium application.
<img src="https://i.imgur.com/qGweFH0.png">

**Comment by bkeating** [comment]
This may just be an overall let down of how dark mode is. I personally expected something ‘flatter’; without the shiny edging. Something more along the lines of atom and it’s One Dark (and many others). Likely why we all noticed. 

Another place this really grinds my gears: the striped finder in list mode. Way too much contrast. 

Then I realize, Apple probably picked their dark mode colors and details based on accessibility-approved balance.  So now I feel like a dick for complaining. I just basically wanted One Dark for all the things.

This is good tho, that dark mode is lame. We still get the bit (light/dark) to hook into and means apps still may want to take it and own it themselves.

Dark mode is dark art.

**Comment by samuelgozi** [comment]
@bkeating I agree it feels like it needs more work, however, I don't believe that they took into account too much the accessibility issues, what leads me to think so is that there are settings for this in the preferences, settings specifically for contrast.

@NordVPN You are right, NordVPN doesn't seem to be an electron app, but that doesn't mean that there is no issue chromium, they probably don't take advantage of dark mode(same as chromium).

**Comment by oomathias** [comment]
I created an empty Application with Xcode and tried some settings. As you can see it's always on and has nothing to do with native or not native.

 - Disabling shadows removes the borders (except the top one).
- Applying a background color does change the border color. Black is almost invisible compared to Electron default one (see below)

I didn't see any other settings while looking at Apple AppKit API.

Code:
```swift
import Cocoa

@NSApplicationMain
class AppDelegate: NSObject, NSApplicationDelegate {

    @IBOutlet weak var window: NSWindow!
    
    func applicationDidFinishLaunching(_ aNotification: Notification) {
        window.hasShadow = false
        window.titlebarAppearsTransparent = true
        window.titleVisibility = .hidden
        window.backgroundColor = NSColor.black
    }

    func applicationWillTerminate(_ aNotification: Notification) {}

}
```
Different settings:
<img width="1410" alt="screenshot" src="https://user-images.githubusercontent.com/465790/46621775-ae684700-cb28-11e8-937e-294efafd70cd.png">

Black bg vs Electron app:
<img width="77" alt="screenshot 1" src="https://user-images.githubusercontent.com/465790/46622098-aceb4e80-cb29-11e8-8906-3d76545e2a65.png">

**Comment by hemlok** [comment]
Thanks for settling that @oomathias!

I guess that means that dark mode support in electron for any transparent applications is going to be a problem 😟

**Comment by oomathias** [comment]
@hemlok A transparent window look like this 😞 
<img width="527" src="https://user-images.githubusercontent.com/465790/46623137-0012d080-cb2d-11e8-85f4-055748185114.png">

**Comment by pronebird** [comment]
@oomathias I have sent a PR that removes the white line for frameless windows.  https://github.com/electron/electron/pull/15012

**Comment by morajabi** [comment]
It is fixed in VS Code Insiders, not yet in the stable release. 
![image](https://user-images.githubusercontent.com/12202757/46738067-11cdb280-ccaa-11e8-996e-35dfc14e42f2.png)

**Comment by MarshallOfSound** [comment]
@morajabi That might be my dark mode PR to vscode being released to insiders, @bpasero has that gone out to insiders now?

**Comment by bpasero** [comment]
@MarshallOfSound yes now it did

**Comment by karlhorky** [comment]
@MarshallOfSound I suppose the PR in question is this one: https://github.com/Microsoft/vscode/pull/59742

Maybe based on this original work: https://github.com/electron/electron/pull/14755

**Comment by Scott-Fischer** [comment]
> I wonder how other already-dark-by-default apps have fared with the move to system-level dark mode support. 

Not well at all. I'm seeing lines across every non-apple dark mode app.

I was so excited for system-wide dark mode that I went against my better judgement and upgraded. This dedicated dark mode actually made my dark mode experience  worse than before when I was just re-skinning electron apps.

**Comment by lehni** [comment]
There's a workaround for this issue, see:

https://github.com/zeit/hyper/issues/3259#issuecomment-431272678

**Comment by karlhorky** [comment]
@lehni this was already mentioned further up in this thread, and does not fix the issue completely: https://github.com/electron/electron/issues/13164#issuecomment-424309847

**Comment by bntzio** [comment]
Hope this gets fixed asap, it's so annoying...

**Comment by brave3d** [comment]
This fix is **only for normal users who are looking for convenience and not a fix for the reported issue**.
It will give Mojave a more similar look to High Sierra where only menubar and dock are dark.

1. Switch to Light mode from **Preferences** -> **General**
2. Enter this command to **terminal** 
`defaults write -g NSRequiresAquaSystemAppearance -bool Yes
`
3. Log out then then log back in.
4. Go to  **Preferences** -> **General** and Switch to Dark Mode

**PS** if you want to undo what you've did repeat the steps with:
`defaults delete -g NSRequiresAquaSystemAppearance`

**Comment by degecko** [comment]
@brave3d This has been recommended already and it does not work. I tested it with multiple apps, including Discord and Spotify, and the lines are still there.

https://i.imgur.com/MKGgu30.png

It's a global effect added on both light and dark modes.

**Comment by brave3d** [comment]
@degecko I've already said that it's **not a fix**, It's will essentially convert the the UI to Light mode, and keep the Dock and menubar in Dark mode so It will be more like High Sierra until a better solution is available.

**Comment by minhnhatspk** [comment]
> **I have a border in frame + transparent**
> 
> <img alt="frame" width="642" src="https://user-images.githubusercontent.com/36361469/46036943-7a942700-c10f-11e8-8612-a42014fcdc56.png">

The same purpose :))
![image](https://user-images.githubusercontent.com/21996261/48193405-fd4e0a00-e37c-11e8-8db6-b3a0ecdf05d4.png)

**Comment by astoilkov** [comment]
I believe this should be closed as it is now possible to implement everything related to Mojave Dark Mode. The following steps are needed:
- add the following to your `Info.plist`
```
<key>NSRequiresAquaSystemAppearance</key>
<false/>
```
- use the `electron.systemPreferences` methods and events related to Mojave Dark Mode like `setAppLevelAppearance` `isDarkMode` and events like `systemPreferences.subscribeNotification('AppleInterfaceThemeChangedNotification'...`

**Comment by gpalomar** [comment]
Not working for me. With this steps, dark mode is applied but topbar shadow is still there. I have a tray application with the typical arrow pointing to it, so my menu must be hidden. Can you give us more information to achieve this? Thanks!

**Comment by kKerem** [comment]
Any other solution?

**Comment by yutent** [comment]
**I get this issue too.** 
```
ENV: 
electron: 4.0.0
os: mac 10.14.2, Light Mode
app: frameless and transparent window
```

**Comment by Razatastic** [comment]
Any solution besides running the "default...-bool no" command?

**Comment by vifird** [comment]
This bug can be fixed by setting`titleBarStyle: 'customButtonsOnHover'`, but causing another question: `titleBar` will appear when hover on it.

**Comment by alvaropinot** [comment]
> This bug can be resolved by setting`titleBarStyle: 'customButtonsOnHover'`, but causing another problem: `titleBar` will appear when hovering on it.

Thanks for that @vifird! But please take into account that it removes the rounded corners too. 😞

**Comment by vifird** [comment]
> > This bug can be resolved by setting`titleBarStyle: 'customButtonsOnHover'`, but causing another problem: `titleBar` will appear when hovering on it.
> 
> Thanks for that @vifird! But please take into account that it removes the rounded corners too. 😞

Yeah, then we must add rounded corners in StyleSheet.

**Comment by Pong420** [comment]
Refer to [electron document](https://electronjs.org/docs/tutorial/mojave-dark-mode-guide)

If you are using [`electron-packager` >= 12.2.0](https://github.com/electron-userland/electron-packager) or [`electron-forge` >= 6](https://github.com/electron-userland/electron-forge) you can set the [`darwinDarkModeSupport`](https://github.com/electron-userland/electron-packager/blob/master/docs/api.md#darwindarkmodesupport) option when packaging and this key will be set for you.

If you are using [`electron-builder` >= 20.37.0](https://github.com/electron-userland/electron-builder) you can set the [`darkModeSupport`](https://www.electron.build/configuration/mac.html) option.

**Comment by MarshallOfSound** [comment]
Heya Folks, quick update on this, I'm going to split this bug into two categories and explain the way forward for each.

### "My app has a white line at the top and it doesn't look right with my custom macOS title bar"

As mentioned above by @Pong420 you should check out our [Dark Mode](https://electronjs.org/docs/tutorial/mojave-dark-mode-guide) guide on the Electron website.  In particular the `darwinDarkModeSupport` flags for packager/forge as that will make the "white line" less obvious and more inline with your OS color scheme when darm mode is enabled.

### "My app is transparent and now it has a white line across the top and looks wrong"

Unfortunately this is currently classified by apple as a "known bug" with Mojave.  I have had discussions with Apple's technical team through TSI's and they agree that the code I provided should work (and does work on <= 10.13).  They have raised this internally and all we can do now is hope that they fix it in a future release of Mojave / macOS.

TLDR: Enable dark mode support for your app to make the white line look slightly better, other than that, this bug is in Apple's court now 👍

**Comment by karlhorky** [comment]
> I have had discussions with Apple's technical team through TSI's and they agree that the code I provided should work (and does work on <= 10.13). They have raised this internally and all we can do now is hope that they fix it in a future release of Mojave / macOS.

@MarshallOfSound awesome, this sounds great! Will you update this thread if there is any movement from Apple's side?

**Comment by vifird** [comment]
> This bug can be resolved by setting`titleBarStyle: 'customButtonsOnHover'`, but causing another problem: `titleBar` will appear when hovering on it.
@MarshallOfSound I'm pretty sure this bug can be fixed, for example provide another value of `titleBarStyle` property, like `customButtonsOnHover` but no hover buttons.

**Comment by Riz-waan** [comment]
Is there anyway to test/do this while under development?

**Comment by vifird** [comment]
@MarshallOfSound 
Any news?
This bug has lasted so long. Why not make a patch on it?
We really really need a correct behavior, although it's Apple' bug.

**Comment by pronebird** [comment]
This can be fixed with the latest patch that landed in the Electron v6 beta 5

https://github.com/electron/electron/issues/15008#issuecomment-497498135

**Comment by vifird** [comment]
@pronebird It's cool, but don't seem like the official resolution?

**Comment by MarshallOfSound** [comment]
Hey folks, I got some good news from apple and validated it just now.  The Catalina beta fixes this issue on Apple's side of things, once Catalina has rolled out as a stable release the white line no longer appears on `transparent: true` or `frame: false` windows 👍

**Comment by bkeating** [comment]
sort of a bummer; I was about to pull the trigger on making some "dark mode causes a white line at the top of electron apps" T-shirt, and have the neck collar have a subtle white line around it. booo.

... not entirely convinced this thread is dead yet. but im excited to hear the good word about Catalina.

**Comment by KevinHughes** [comment]
#18485 has been backported to Electron 5.0.3, and it fixes this issue for me (using `customButtonsOnHover` and the appropriate flags)! I had a feeling that a patch like this could be done... a big thank you to @codebytere, now I can start beta testing my app for real

**Comment by tvvxaop** [comment]
This also existed for light mode on my MacOS. 

Confirmed fixed after upgrading to Catalina.

**Comment by nickensoul** [comment]
Catalina 10.15, seems like adds new issues – strange black (or almost) border and another lighter border (likely it was before; Developer Tools window, for example, has this frame too) appears after updating.

Dark border
![image](https://user-images.githubusercontent.com/14836054/67206480-64dc5880-f41a-11e9-8bf8-b42475c76eca.png)
Window's options list:
```js
{
  show: false,
  darkTheme: true,
  backgroundColor: '#000000ff',
  frame: false,
  titleBarStyle: 'hiddenInset',
  vibrancy: 'dark'
}
```

Lighter border
![image](https://user-images.githubusercontent.com/14836054/67206518-77569200-f41a-11e9-971f-cc4f1b96f2d9.png)
Window's options list:
```js
{
  show: false,
  center: true,
  darkTheme: true,
  resizable: false,
  maximizable: false,
  fullscreenable: false,
  frame: false,
  backgroundColor: '#1f1f1f',
  titleBarStyle: 'hiddenInset'
}
```

How to just remove these chrome forever?

**Comment by bernatfortet** [comment]
@nickensoul did you figure out how to get rid of the black line?

**Comment by nickensoul** [comment]
@bernatfortet nope, just left as it is for now. I believe it requires some changes of interactions with system APIs (and I think, it's the Electron core, so would need a fork)
So, still waiting.

**Comment by bernatfortet** [comment]
gotcha, thanks!

**Comment by msavin** [comment]
The white line on top of the screen is transparent. If you can put a `#000000` line beneath it,  you can dim it quite a bit.

**Comment by msavin** [comment]
The white line on top of the screen is transparent. If you can put a `#000000` line beneath it,  you can dim it quite a bit.

It's a bit tricky to get it to render, but adding the following to my custom titleBar did the trick.

```css
border-top: 1px solid #000000
padding-top: 1px
margin-top: -1px
````

The white bar came down from `rgb(175,175,175)` to `rgb(151,151,151)`

**Comment by adam-lynch** [comment]
For those coming to this now, it seems to be fixed in Catalina. I just tested it on 10.15.2

**Comment by sergeyfilimonov** [comment]
> For those coming to this now, it seems to be fixed in Catalina. I just tested it on 10.15.2

In Atom I still see the line:
https://user-images.githubusercontent.com/43918077/67483562-bda42f00-f66e-11e9-8fb4-4590dc06822e.png

It seems Atom is using an older version of Electron...

**Comment by hologerry** [comment]
on Big Sur all the windows have white line

**Comment by robjtede** [comment]
It's a black border outside an off-white border, except on Discord it currently fills in the corners too :(

<img width="300" alt="2020-06-24_01 27 52-886a2eca" src="https://user-images.githubusercontent.com/3316789/85481614-0d4e4800-b5ba-11ea-9c3b-6b84d47dcae0.png">

**Comment by MarshallOfSound** [comment]
@robjtede I haven't spun up the developer beta yet, but if that's happening with all Electron apps make sure you raise it in the feedback assistant 😄

**Comment by hologerry** [comment]
> @robjtede I haven't spun up the developer beta yet, but if that's happening with all Electron apps make sure you raise it in the feedback assistant 😄

Not only the third party apps have this issue, but also all the Apple apps such as Finder, Music, Safari do.
Here is the screenshot:
<img width="1680" alt="Screen Shot 2020-06-25 at 00 04 53" src="https://user-images.githubusercontent.com/11660168/85591122-9aa3a200-b677-11ea-92cb-5371ec61ba4d.png">

I raise the issue here is to make more people aware of this Big Sur's bug, so that the Apple team could notice it and fix it.
Of course, I have raise the issue on feedback assistant and the developer forum.

**Comment by DoppiaeSSe** [comment]
> > @robjtede I haven't spun up the developer beta yet, but if that's happening with all Electron apps make sure you raise it in the feedback assistant 😄
> 
> Not only the third party apps have this issue, but also all the Apple apps such as Finder, Music, Safari do.
> Here is the screenshot:
> <img alt="Screen Shot 2020-06-25 at 00 13 13" width="548" src="https://user-images.githubusercontent.com/11660168/85592482-cd9a6580-b678-11ea-8597-7a633ae64d9b.png">
> 
> I raise the issue here is to make more people aware of this Big Sur's bug, so that the Apple team could notice it and fix it.
> Of course, I have raise the issue on feedback assistant and the developer forum.
> 
> Edit: Update the screenshot to make it bigger.

Any news? It's kinda annoying seeing this white lines in every single window when in dark mode...

**Comment by hologerry** [comment]
Looks like only the Electron-based apps still have this issue on macOS Big Sur, for example, VS Code.
System apps and Chrome have fixed this issue.
If you have not noticed this issue, just download iTerm and VS Code, set dark themes for both, look at the top of the apps.

system appearance Light:
![image](https://user-images.githubusercontent.com/11660168/120377359-9ab02580-c34f-11eb-9608-c97c3e629efb.png)

system appearance Dark:
![image](https://user-images.githubusercontent.com/11660168/120377398-a6035100-c34f-11eb-8b87-b6fd24dee889.png)

IT IS SO ANNOYING!

**Comment by johnnyshankman** [comment]
Why was this closed? I still have this issue clear as day. By default this is an issue.

<img width="460" alt="Screenshot 2023-12-31 at 12 43 44 PM" src="https://github.com/electron/electron/assets/6632701/8f12db55-f876-4108-8708-cc31684908f7">

**Comment by FireMapleLeaves** [comment]
Electron 20 + Mac OS Sonoma 14.5  still has this issue

**Comment by AlwaysExtraordinary** [comment]
macOS 26 still has this problem... 

<img width="2252" height="1630" alt="Image" src="https://github.com/user-attachments/assets/2cd8cc2f-29c8-4ab0-a132-63131032f065" />

**Comment by dMcCaffree** [comment]
@AlwaysExtraordinary I was able to fix this in my app by setting `transparent: true` and `frame: false` in `BrowserWindow()`.

**Comment by ueoo** [comment]
Looks like macOS 27 beta 3 finally removed this white line. 

<img width="1119" height="139" alt="Image" src="https://github.com/user-attachments/assets/53558beb-8e8c-4850-b24c-4c2234d4aa22" />
